src/run.jl

```diff
@@ -53,19 +52,42 @@ function spawn_sandboxed_julia(julia::VersionNumber, args=``; interactive=true,
 end
 
 """
-    run_sandboxed_test(julia::VersionNumber, pkg; do_depwarns=false, log_limit=5*1024^2, time_limit=60*45)
+    run_sandboxed_test(julia::VersionNumber, pkg; do_depwarns=false, log_limit=5*1024^2,
+                       time_limit=60*45)
 
-Run the unit tests for a single package `pkg` inside of a sandbox using the Julia version
-`julia`. If `do_depwarns` is `true`, deprecation warnings emitted while running the package's
-tests will cause the tests to fail. Test will be forcibly interrupted after `time_limit`
-seconds or if the log becomes larger than `log_limit`.
+Run the unit tests for a single package `pkg` inside of a sandbox using Julia version
+`julia`. If `do_depwarns` is `true`, deprecation warnings emitted while running the
+package's tests will cause the tests to fail. Test will be forcibly interrupted after
+`time_limit` seconds or if the log becomes larger than `log_limit`.
 
 A log for the tests is written to a version-specific directory in the NewPkgEval root
 directory.
+
+Refer to `run_sandboxed_julia`[@ref] for more possible `keyword arguments.
 """
-function run_sandboxed_test(julia::VersionNumber, pkg::String; log_limit = 5*1024^2 #= 5 MB =#,
+function run_sandboxed_test(julia::VersionNumber, pkg; log_limit = 5*1024^2 #= 5 MB =#,
                             time_limit = 45*60, do_depwarns=false, kwargs...)
-    mkpath(log_path(julia))
+    # everything related to testing in Julia: version compatibility, invoking Pkg, etc
+
+    if pkg.name in skip_lists[pkg.registry]
+        return missing, :skip, :explicit, missing
+    end
+
+    # can we even test this package?
```

Comment:
@KristofferC is this a good way to check if (any version of) a package supports a specific Julia version? Couldn't find an API. All Compat.toml's in General seem to contain a Julia bound, but I wasn't sure if this is a requirement.

Reply:
Compat info is "restrictive" in the sense that the absence of any compat info of a dependency means that the package is compatible with all versions of that dependency. So `supported` should default to `true`.

Out of curiosity, what packages have upper bounded Julia to 1.x?

Reply:
> So `supported` should default to `true`.

But I never set it to false, only to true if I find a VersionRange that matches. Wasn't there a requirement to have at least `julia = 1` compat (i.e. there will never be a package without a `julia` compat bound were we would need to be conservative and default to true)? All packages seem to have a bound for `julia`.


> Out of curiosity, what packages have upper bounded Julia to 1.x?

The JLL packages have a compat on 1.3, for one.

Reply:
> But I never set it to false, only to true if I find a VersionRange that matches

I meant it is initially set to `false` so it would still be `false` if there wasn't a julia entry?

> Wasn't there a requirement to have at least julia = 1 compat (i.e. there will never be a package without a julia compat bound were we would need to be conservative and default to true)? All packages seem to have a bound for julia.

I know that I had some errors in my code before that assumed that there was a julia entry in Compat but I think it was like one package and that one got fixed.

> The JLL packages have a compat on 1.3, for one.

True, should we run these at all on PkgEval?

Reply:
> True, should we run these at all on PkgEval?

Not necessarily, but it was just an example, Turns out there's quite some packages that have a lower-bound on Julia 1.1 though.

> I meant it is initially set to `false` so it would still be `false` if there wasn't a julia entry?

Yeah, but by switching it to `true` initially that would just defeat the entire thing as it's implemented right now :stuck_out_tongue: I'll have a look at making it more correct though.

Reply:
https://github.com/JuliaComputing/NewPkgEval.jl/pull/30/commits/bbfbac5f0d097463cbf7e934a7422527b9ed10eb should be correct